### PR TITLE
[oneapi] Introduce `nnfw_set_input_tensorinfo`

### PR DESCRIPTION
**Pull request by wateret**
Introduce a new API function `nnfw_set_input_tensorinfo`. This will
replace `nnfw_apply_tensorinfo`.

Reasons why introducing a new API:

- The name may be misleading - it can change inputs' info, not all
  tensors
- Function signature - it gets `tensor_info` by value
- Different behavior depend on the session state

ONE-DCO-1.0-Signed-off-by: Hanjoung Lee <hanjoung.lee@samsung.com>

### CODE REVIEW

**Comment by wateret** [comment]
Context : #1423

File: runtime/onert/api/include/nnfw.h [hunk 1]
```diff
@@ -202,6 +204,28 @@ NNFW_STATUS nnfw_load_model_from_file(nnfw_session *session, const char *package
...
+ * @brief    Set input model's tensor info for resizing
+ *
+ * This function can be called in any time after calling {@link nnfw_model_load_from_file}. Changing
+ * input tensor's shape will cause shape inference for the model. There are two different types of
+ * shape inference - static and dynamic. Which one to use is depend on the current state of the
+ * session.
+ * When it is called after calling {@link nnfw_model_load_from_file} and before calling {@link
+ * nnfw_prepare}, this info will be used when {@link nnfw_prepare}. And it will perform static shape
+ * inference for all tensors.
+ * When it is called after calling {@link nnfw_prepare} or even after {@link nnfw_run}, this info
```
Comment (hyunsik-yoon) [inline runtime/onert/api/include/nnfw.h:217]: Does ` even after {@link nnfw_run}` means that this call will make an effect on the next run?

Comment (wateret) [reply runtime/onert/api/include/nnfw.h:217]: Yes!

File: runtime/onert/api/include/nnfw.h [hunk 2]
```diff
@@ -202,6 +204,28 @@ NNFW_STATUS nnfw_load_model_from_file(nnfw_session *session, const char *package
 NNFW_STATUS nnfw_apply_tensorinfo(nnfw_session *session, uint32_t index,
                                   nnfw_tensorinfo tensor_info);
 
+/**
+ * @brief    Set input model's tensor info for resizing
+ *
+ * This function can be called in any time after calling {@link nnfw_model_load_from_file}. Changing
```
Comment (chunseoklee) [inline runtime/onert/api/include/nnfw.h:210]:
```suggestion
 * This function can be called at any time after calling {@link nnfw_model_load_from_file}. Changing
```

**Pull request merged**